src/code/views/node-view.coffee

```diff
@@ -49,10 +59,12 @@ NodeTitle = React.createFactory React.createClass
 
   renderTitleInput: ->
     displayTitle = @displayTitleForInput(@props.title)
+    canDeleteWhenEmpty = @props.node.addedThisSession and not @titleUpdated
     (input {
       type: "text"
       ref: "input"
       className: "node-title"
+      onKeyUp: if canDeleteWhenEmpty then @detectDeleteWhenEmpty else null
```

Comment:
I still don't get why we need `canDeleteWhenEmpty`, when you've got to test `@titleUpdated` twice anyway, instead of just checking `@props.node.addedThisSession and not @titleUpdated` in the handler, but it's not a big problem in any case.

I think this is fine in general, and works well.